client/src/main/java/io/pravega/client/tables/KeyVersion.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables;
+
+import io.pravega.client.tables.impl.TableSegmentKeyVersion;
+import java.io.Serializable;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Version of a Key in a Table.
```

Comment:
I find this notion of "key version" confusing. I understand you want the comparison to be on separate value compared to making it conditional on the value itself, and I think I understand why you associate the version to the key: you the key is what is constant in a key-value pair. But, the version really refers to the KV pair, and it indicates state changes to a particular KV pair, where the state change is reflected in the change of a value.

I'm saying all this to ask whether we can disassociate version from key. If we call it just version, then it would be ideal but it is too generic maybe and might conflict with some other definition of version. Another option is to call it `KVVersion` to associate to a KV pair.

Let me know what you think.

Reply:
The KVT is a collection of KV pairs, then each KV pair is essentially immutable. As such it cannot have "versions". 

The KVT can also be viewed as a collection of Keys that have values associated, and every KVT mutation modifies the Value associated with a Key. As such, every such mutation creates a new KVP for that Key, which means the version of the Key needs to change. Hence the association of Versions to Keys (i.e., `KeyVersion`).

Reply:
In the general case of a table, a triple {row, column, version} gives the value of a cell, e.g., as in hbase:

https://hbase.apache.org/book.html#versions

if we treat KV pairs as a special case of a table with a single column, then {key, version} gives the corresponding value of that version. I think that's why you say that the version is associated to the key, but if we think more generally about tables, then the version is really about the cell and its value.

I´d rather not call it `KeyVersion` because the key is a coordinate to the value, and the version is not about the coordinate. But, I don't fully understand whether that breaks the internal abstractions (e.g., `TableSegment`), in which case we might not have a choice. Let me know.

Reply:
This is not a multi-column table. It's a 2-column table, one column being key and the other value. We cannot use `version` to execute a lookup (i.e., get version X of key Y). We can only get the latest version of a Key. Versions are only there to enable conditional updates.

This API does nothing to change the meaning of versions. Versions are defined like this in Table Segments and the KeyValue Table API is just passing them through and exposing them. 

If you are looking to make such a semantic change, we'd have to likely redo most of the Table Segment work, which is out of scope for this project.

Reply:
If we can call it `Version` or add a prefix to version that doesn't make it look like we are versioning the key rather than the value, I'm good. Internally, the version can remain implemented as is, so no work out of scope. Does it work?

Reply:
Ok. I renamed it to `Version`.